src/components/DatePicker/index.ios.js

```diff
@@ -20,124 +19,117 @@ const datepickerPropTypes = {
     ...keyboardStatePropTypes,
 };
 
-class DatePicker extends React.Component {
-    constructor(props) {
-        super(props);
+function DatePicker(props) {
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+    const [selectedDate, setSelectedDate] = useState(props.value || props.defaultValue ? moment(props.value || props.defaultValue).toDate() : new Date());
+    const initialValue = useRef(null);
+    const inputRef = useRef(null);
 
-        this.state = {
-            isPickerVisible: false,
-            selectedDate: props.value || props.defaultValue ? moment(props.value || props.defaultValue).toDate() : new Date(),
-        };
-
-        this.showPicker = this.showPicker.bind(this);
-        this.reset = this.reset.bind(this);
-        this.selectDate = this.selectDate.bind(this);
-        this.updateLocalDate = this.updateLocalDate.bind(this);
-    }
-
-    showPicker() {
-        this.initialValue = this.state.selectedDate;
+    const showPicker = useCallback(() => {
+        initialValue.current = selectedDate;
 
         // Opens the popover only after the keyboard is hidden to avoid a "blinking" effect where the keyboard was on iOS
         // See https://github.com/Expensify/App/issues/14084 for more context
-        if (!this.props.isKeyboardShown) {
-            this.setState({isPickerVisible: true});
+        if (!props.isKeyboardShown) {
+            setIsPickerVisible(true);
             return;
         }
         const listener = Keyboard.addListener('keyboardDidHide', () => {
-            this.setState({isPickerVisible: true});
+            setIsPickerVisible(true);
             listener.remove();
         });
         Keyboard.dismiss();
-    }
+    }, [props.isKeyboardShown, selectedDate]);
+
+    useEffect(() => {
+        const el = inputRef.current;
+
+        if (!isFunction(props.innerRef)) {
+            return;
+        }
+
+        if (el && el.focus && typeof el.focus === 'function') {
+            props.innerRef({ ...el, focus: showPicker });
+            return;
+        }
+
+        props.innerRef(el);
+    }, [props, props.innerRef, showPicker]);
```

Comment:
listening to the whole `props` can cause unneeded rerenders, just put as dependencies the one you use